lib/src/realtime_client.dart

```diff
@@ -222,28 +227,28 @@ class RealtimeClient {
       message.payload,
     );
 
-    if (isConnected()) {
+    if (isConnected) {
       callback();
     } else {
       sendBuffer.add(callback);
     }
   }
 
   /// Returns the URL of the websocket.
-  String endPointURL() {
+  String get endPointURL {
     final params = Map<String, String>.from(this.params);
     params['vsn'] = Constants.vsn;
     return _appendParams(endPoint, params);
   }
 
   /// Return the next message ref, accounting for overflows
   String makeRef() {
-    int newRef = ref + 1;
-    if (newRef < 0) {
-      newRef = 0;
+    final int newRef = ref + 1;
+    if (newRef == ref) {
+      ref = 0;
+    } else {
+      ref = newRef;
```

Comment:
I did not know this, so please correct me if I'm wrong, but it seems like the behavior of integer going above the max value in js and Dart are different. js will actually add 1 more number to the max where as Dart will go to the smallest possible integer, so it is probably okay to keep inside the if as the original one. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER